packages/clarity-js/src/layout/dom.ts

```diff
@@ -176,6 +179,18 @@ export function update(node: Node, parent: Node, data: NodeInfo, source: Source)
 
         // Update selector
         updateSelector(value);
+
+        // if updated node is a part of fragment and the fragment is not being tracked currently, schedule a mutation on the fragment node
```

Comment:
>if updated node is a part of fragment [](http://example.com/codeflow?start=11&length=37)

Would it be possible to do this only for the "root" target node of the mutation and not every child node? If so, we can eliminate the cross-talk between dom & mutation and just schedule fragment node mutation from within mutation module while keeping the update fragment logic simple.